cmd/ci-operator-prowgen/main.go

```diff
@@ -84,7 +107,7 @@ func (o *options) process() error {
 // Generate a PodSpec that runs `ci-operator`, to be used in Presubmit/Postsubmit
 // Various pieces are derived from `org`, `repo`, `branch` and `target`.
 // `additionalArgs` are passed as additional arguments to `ci-operator`
-func generatePodSpec(info *config.Info, target string, additionalArgs ...string) *kubeapi.PodSpec {
+func generatePodSpec(info *config.Info, target string, cmd, args []string, additionalArgs ...string) *kubeapi.PodSpec {
```

Comment:
I don't like how this this makes all default case callsites cluttered with `nil`, `nil`. I like the advice in https://dave.cheney.net/practical-go/presentations/qcon-china.html#_design_apis_for_their_default_use_case. I think the only caller that cares about `cmd` and `args` (`generatePodSpecRandom`) can overwrite the `Containers[0].Command` and `Containers[0].Args` fields of the returned podspec, eliminating the impact on other callsites.

Reply:
Heh, I spent ~5ms thinking about the interface. Agreed, I will change it to something more decent.